scripts/Dockerfile

```diff
@@ -21,45 +21,44 @@ ENV CMAKE_C_COMPILER=/usr/bin/gcc
 ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV SUNDIALS_INST=/home/pybamm/.local
-ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib:
+ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
+
+RUN conda create -n pybamm python=3.11
+RUN conda init --all
+SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
+RUN conda install -y pip
 
 ARG IDAKLU
 ARG ODES
 ARG JAX
 ARG ALL
 
-RUN conda create -n pybamm python=3.9
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+RUN pip install --upgrade --user pip setuptools wheel wget cmake
 
 RUN if [ "$IDAKLU" = "true" ]; then \
-    pip install --upgrade --user pip setuptools wheel wget && \
-    pip install cmake==3.22 && \
     python scripts/install_KLU_Sundials.py && \
     git clone https://github.com/pybind/pybind11.git && \
-    pip install --user -e ".[all,dev]"; \
+    pip install --user -e ".[all,dev,docs]"; \
     fi
 
 RUN if [ "$ODES" = "true" ]; then \
-    pip install cmake==3.22 && \
-    pip install --upgrade --user pip wget && \
     python scripts/install_KLU_Sundials.py && \
-    pip install --user -e ".[all,odes,dev]"; \
+    git clone https://github.com/pybind/pybind11.git && \
+    pip install --user -e ".[all,dev,docs,odes]"; \
     fi
 
 RUN if [ "$JAX" = "true" ]; then \
-    pip install --user -e ".[jax,all,dev]";\
+    python scripts/install_KLU_Sundials.py && \
+    git clone https://github.com/pybind/pybind11.git && \
```

Comment:
This isn't required for JaxSolver
```suggestion
```

Reply:
I agree, I added it because Jax and scikits.odes are optional, but it would be good to have IDAKLU by default so that one would get IDAKLU + Jax. Do we want the optional solvers to be exclusive?

Reply:
Hmm yes, but building pybamm with KLU locally would require extra time, should be okay if a developer fetches it from Dockerhub. We could make the IDAKLU solver as a default solver here.

Reply:
The only issue would be an increase in the build time, for us (which is okay because images won't be built on pull requests), but we would have to update the docs about this change. At this time, your idea sounds more reasonable because the use of `IDAKLU=true` _and_ `JAX=true` is more explicit about what solver components are being installed, compared to the case if someone were to get `IDAKLU` by default with the Jax solver (one build argument is installing both, which is slightly misleading to the user).